File: Advanced C#/Data-Structures/StacksAndQueues/StacksAndQueues/Queue.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace StacksAndQueues
{
    public class Queue
    {
        public Node Front { get; set; }
        private Node Rear { get; set; }

        public Queue()
        {
            Rear = Front;
        }

        public void Enqueue(string value)
        {
            Node node = new Node(value);

            if (Front == null)
            {
                Front = node;
                Rear = node;
            }
            else
            {

                Rear.Next = node;
                Rear = node;

            }
        }

        public string Dequeue()
        {
            if (Front == null)
            {
                throw new Exception("Empty queue");
            }
            string result = Front.Value;
            Front = Front.Next;

            return result;
        }

        public string Peek()
        {
            if (Front == null)
            {
                throw new Exception("Empty queue");
            }
            return Front.Value;
        }

    }
    }

File: Advanced-Javascript/data-structure/StacksandQueues/queue.js
'use strict';

const Node = require('../linked-list/node.js');

class Queue {
    constructor(){
        this.front = null;
        this.rear = this.front;
    }

    enqueue(value){
        let node = new Node(value);

        if(this.front === null){
            this.front = node;
            this.rear = node;
        }
        else {
            this.rear.next = node;
            this.rear = node;
        }
    }

    dequeue(){
        if(this.front === null){
            return null;
        }
        let result = this.front.value;
        this.front = this.front.next;
        return result;
    }

    peek(){
        if(this.front === null){
            return null;
        }
        return this.front.value;
    }
}

module.exports = Queue;